src/simulator_core/solver/matrix/matrix.py

```diff
@@ -106,20 +96,32 @@ def get_solution(self, index: int, number_of_unknowns: int) -> list[float]:
         be provided
         :return: list with teh solution
         """
-        #  TODO add checks if the index and number of unknowns are within the range of the solution.
-        return self.sol_new[index:index + number_of_unknowns]
-
-    def dump_matrix(self, file_name: str = 'dump.csv') -> None:
+        if index > self.num_unknowns:
+            raise IndexError(f"Index ({index}) greater than number "
+                             f"of unknowns ({self.num_unknowns})")
+        if (index + number_of_unknowns) > self.num_unknowns:
+            raise IndexError(f"Index ({index}) plus request number of elements "
+                             f"({number_of_unknowns}) greater than "
+                             f"number of unknowns {self.num_unknowns}")
+        result: list[float] = self.sol_new[index:index + number_of_unknowns].tolist()
```

Comment:
Why define result: list[float] when we know that the output is already list[float]?